plugins/enthusiast-common/enthusiast_common/agents/builder.py

```diff
@@ -0,0 +1,199 @@
+from dataclasses import fields, dataclass
+
+from langchain_core.language_models import BaseLanguageModel
+from langchain_core.prompts import PromptTemplate, ChatMessagePromptTemplate
+from langchain_core.tools import BaseTool
+
+from .base import BaseAgent
+from ..config import AgentConfig, LLMConfig, AgentConfigValidator
+from ..injectors import BaseProductRetriever, BaseDocumentRetriever, BaseInjector
+from ..services.conversation import ConversationService
+from ..tools import BaseAgentTool, BaseLLMTool, BaseFunctionTool
+from ..registry import BaseLanguageModelRegistry, BaseEmbeddingProviderRegistry, BaseDBModelsRegistry
+from ..repositories import (
+    BaseProductRepository,
+    BaseDataSetRepository,
+    BaseDocumentChunkRepository,
+    BaseUserRepository,
+    BaseMessageRepository,
+    BaseConversationRepository,
+)
+
+
+@dataclass
+class RepositoriesInstances:
+    user: BaseUserRepository
+    message: BaseMessageRepository
+    conversation: BaseConversationRepository
+    data_set: BaseDataSetRepository
+    document_chunk: BaseDocumentChunkRepository
+    product: BaseProductRepository
+
+
+class AgentBuilder:
```

Comment:
This looks cool! Would it be something to live inside the server, that would also be able to read agent's definition (how an admin/developer configured the agent)?